### ISSUE TITLE
valet-fpm.conf # treated differently than ;

### ISSUE THREAD

**Issue by grantholle** [laravel/valet#896]
I did a huge `brew upgrade` and got php 7.4. After this, I did a `composer global update`. I'm getting 502's on everything, the issue being:

```
*1 connect() to unix:/Users/grant/.config/valet/valet.sock failed (2: No such file or directory) while connecting to upstream
```

Things I've tried:

- `valet install` a hundred times
- `valet uninstall` and then `valet install`
- `valet use php` (which is 7.4)
- brew uninstalled php, dnsmasq, and nginx, reinstalled
- removed the global valet package, reinstalled
- `brew doctor`'d everything. All good.

Read the following posts and tried their solutions:

- https://laracasts.com/discuss/channels/laravel/502-bad-gateway-nginx-laravel-valet?page=1#reply=560368
- https://laracasts.com/discuss/channels/code-review/laravel-valet-broken-after-upgrading-to-php-74?page=1#reply=566240

Thanks for all your help!

**Comment by grantholle** [comment]
It might be related to my php-fpm being screwed up?

```
[06-Jan-2020 10:44:08] ERROR: [/usr/local/etc/php/7.4/php-fpm.d/valet-fpm.conf:11] value is NULL for a ZEND_INI_PARSER_ENTRY
[06-Jan-2020 10:44:08] ERROR: Unable to include /usr/local/etc/php/7.4/php-fpm.d/valet-fpm.conf from /usr/local/etc/php/7.4/php-fpm.conf at line 11
[06-Jan-2020 10:44:08] ERROR: failed to load configuration file '/usr/local/etc/php/7.4/php-fpm.conf'
[06-Jan-2020 10:44:08] ERROR: FPM initialization failed
```

I've uninstalled and `brew install php --build-from-source`.

**Comment by grantholle** [comment]
I've found that the issue is in the auto-generated `valet-fpm.conf` file found in `/usr/local/etc/php/7.4/php-fpm.d/valet-fpm.conf` that using `#` as a commented line causes problems.

Changing

```
# when uncommented, these values will take precedence over settings declared elsewhere
```

to 

```
; when uncommented, these values will take precedence over settings declared elsewhere
```

causes fpm to start successfully, therefore getting the sock file to generate and things start up as expected.

**Comment by grantholle** [comment]
Just saw #894 which will fix this...

**Comment by drbyte** [comment]
Was closed by #894
